lib/py/tests/edge_context_tests.py

```diff
@@ -349,6 +353,8 @@ def test_service_auth(self):
         self.assertEqual(request_context.service.on_behalf_of_id, None)
         self.assertEqual(request_context.service.on_behalf_of_roles, None)
         self.assertEqual(request_context.service.requests_elevated_access, None)
+        with self.assertRaises(NoAuthenticationError):
```

Comment:
Maybe add a test with a user ID that is longer than 9 characters?

Reply:
the issue here was that we were incorrectly characterizing this as service auth - it just so happened that the checks in place would only propogate the issue on longer user ids.